fxa/profile.py

```diff
@@ -0,0 +1,73 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+from six import string_types
+
+from fxa.errors import OutOfProtocolError
+from fxa._utils import APIClient, BearerTokenAuth
+
+
+DEFAULT_SERVER_URL = "https://profile.accounts.firefox.com/v1"
+VERSION_SUFFIXES = ("/v1",)
+DEFAULT_CACHE_EXPIRY = 300
+
+
+class Client(object):
+    """Client for talking to the Firefox Accounts Profile server"""
+
+    def __init__(self, server_url=None):
+        if server_url is None:
+            server_url = DEFAULT_SERVER_URL
+        if not isinstance(server_url, string_types):
+            self.apiclient = server_url
+        else:
+            server_url = server_url.rstrip('/')
+            if not server_url.endswith(VERSION_SUFFIXES):
+                server_url += VERSION_SUFFIXES[0]
+            self.apiclient = APIClient(server_url)
+
+    @property
+    def server_url(self):
+        return self.apiclient.server_url
+
+    def get_profile(self, token):
+        """Get all profile data for the user associated with this token."""
+        url = '/profile'
+        resp = self.apiclient.get(url, auth=BearerTokenAuth(token))
+
+        for field in ("uid", "email"):
```

Comment:
Good catch, we do indeed get the avatar.  The profile-server returns an "avatar" field but its API docs are out of date and do not mention it.